file/builder.go

```diff
@@ -598,13 +602,31 @@ func (b *stateBuilder) routes() {
 		return
 	}
 
+	unsupportedRoutes := []string{}
 	for _, r := range b.targetContent.Routes {
 		r := r
 		if err := b.ingestRoute(r); err != nil {
 			b.err = err
 			return
 		}
 	}
+
+	// check routes' paths format
+	if b.checkRoutePaths {
+		allRoutes, err := b.intermediate.Routes.GetAll()
+		if err != nil {
+			b.err = err
+			return
+		}
+		for _, r := range allRoutes {
+			if utils.HasPathsWithRegex300AndAbove(r.Route) {
+				unsupportedRoutes = append(unsupportedRoutes, *r.Route.ID)
```

Comment:
Please ensure that ID cannot be nil here. I'm not sure.

Reply:
ingested routes always have an ID: https://github.com/Kong/deck/blob/main/file/builder.go#L772-L796